lms/resources/_js_config.py

```diff
@@ -202,6 +230,40 @@ def _debug(self):
 
         return debug_info
 
+    def _get_students(self):
+        """
+        Return the list of student dicts for the request.
+
+        Returns one student dict for each student who has launched the
+        assignment and had grading info recorded for them.
+        """
+        grading_infos = self._grading_info_service.get_by_assignment(
+            oauth_consumer_key=self._request.lti_user.oauth_consumer_key,
+            context_id=self._request.params.get("context_id"),
+            resource_link_id=self._request.params.get("resource_link_id"),
+        )
+
+        # The list of "student" dicts that we'll return.
+        students = []
+
+        # Create a "student" dict for each GradingInfo.
+        for grading_info in grading_infos:
+            h_user = HUser(
+                authority=self._request.registry.settings["h_authority"],
+                username=grading_info.h_username,
+                display_name=grading_info.h_display_name,
+            )
+            students.append(
+                {
+                    "userid": h_user.userid,
+                    "displayName": h_user.display_name,
+                    "LISResultSourcedId": grading_info.lis_result_sourcedid,
+                    "LISOutcomeServiceUrl": grading_info.lis_outcome_service_url,
+                }
+            )
```

Comment:
Same feedback as before here. If we yielded out the student and wrapped the call with `list` this would be a little snappier.

Reply:
Would this really make it faster? I'm not sure I understand how yielding the values and then turning them into a list right after is faster than just returning the list in the first place?

Reply:
Snappier as in short and sweet, rather than as in code execution-wise

Reply:
https://github.com/hypothesis/lms/pull/1547/commits/eb705f6f3d1dca38aeb2966f121e4b04f9442797